schedulearn-frontend/src/components/Login/LoginPage.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import dotnetify, { dotnetifyVM } from "dotnetify";
+import { UsernameContext } from "../Contexts/UsernameContext";
+import { UsernameState } from "../Contexts/UsernameContext";
+
+dotnetify.hubServerUrl = "http://localhost:5000";
+
+interface State {
+  CurrentUser?: User;
+  VM?: dotnetifyVM;
+  UserName: string;
+}
+
+interface User {
+  Id: number;
+  Name: string;
+  Surname: string;
+  JobTitleId: number;
+}
+
+export default class LoginPage extends React.Component<{}, State> {
+  constructor(props: {}) {
+    super(props);
+    this.state = { CurrentUser: { Id: 0, Name: "name", Surname: "surname", JobTitleId: 0 }, UserName: "User name" };
+  }
+
+  componentDidMount(): void {
+    const vm = dotnetify.react.connect("UserBaseVM", this);
+    this.setState({ VM: vm });
+  }
+
+  componentWillUnmount(): void {
+    if (this.state.VM !== undefined)
+      this.state.VM.$destroy();
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  $dispatch = (iValue: any): void => {
+    if (this.state.VM !== undefined)
+      this.state.VM.$dispatch(iValue);
+  };
+
+  handleUsername = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    this.setState({ UserName: event.target.value });
+  }
+
+  handleSetUsername = (context: UsernameState): void => {
+    this.$dispatch({ SetCurrentUser: { Name: this.state.UserName } });
+    context.setUsername(this.state.UserName);
```

Comment:
As mentioned before, just setting the UserName won't work in this case, since we need the user Id and might need other fields eventually as well, but the main problem is is that dispatch does not immediately return this. The workaround for this is to probably introduce a "componentDidUpdate" function, which can pass previousProps and previousState as function arguments. Using this, you can compare if your currentUser is different from the previously set user, and if it is, you can set it onto the context.

Reply:
Implemented the proposed solution.